### PR TITLE
Wireframe view

### PR DESCRIPTION
**Pull request by BagelOrb**
Changes needed to support edge-overhang for wireframe view.

Note that I haven't been able to get edge-overhang to work, so this PR is not needed for the linked Cura PR.

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
Devs, see CURA-3849.

File: UM/View/RenderBatch.py [hunk 1]
```diff
@@ -288,10 +292,12 @@ def _renderItem(self, item):
...
         if mesh.hasIndices():
             if self._render_range is None:
+                element_count = mesh.getFaceCount();
                 if self._render_mode == self.RenderMode.Triangles:
-                    self._gl.glDrawElements(self._render_mode, mesh.getFaceCount() * 3 , self._gl.GL_UNSIGNED_INT, None)
-                else:
-                    self._gl.glDrawElements(self._render_mode, mesh.getFaceCount(), self._gl.GL_UNSIGNED_INT, None)
+                    element_count = mesh.getFaceCount() * 3
+                elif self._render_mode == self.RenderMode.TrianglesAdjacency:
+                    element_count = mesh.getFaceCount() * 3 * 4
```
Comment (LipuFei) [inline UM/View/RenderBatch.py]: Shall we still have the `else` at the end? Then other cases won't be affected. Like this:
```
else:
    element_count = mesh.getFaceCount()
```

File: UM/View/Renderer.py [hunk 2]
```diff
@@ -12,11 +12,6 @@
 #   objects should be rendered but the actual rendering process happening after a
 #   sorting step.
 class Renderer():
-    RenderTriangles = 1
```
Comment (LipuFei) [inline UM/View/Renderer.py]: Why are those removed btw?

Comment (LipuFei) [inline UM/View/Renderer.py]: Ah. We now use `RenderMode`.

File: UM/View/RenderBatch.py (hunk 1 above)
Comment (LipuFei) [inline UM/View/RenderBatch.py]: Ok, that's done before the `if`s. Missed that part
